packages/gestalt/src/useGetChildrenToArray.js

```diff
@@ -0,0 +1,59 @@
+// @flow strict
+import { cloneElement, Children, type Node } from 'react';
+
+const ALLOWED_CHILDREN_MAP = {
+  'main': ['SideNavigation.Section', 'SideNavigation.TopItem', 'SideNavigation.Group'],
+  'nested': ['SideNavigation.NestedItem', 'SideNavigation.NestedGroup'],
+};
+
+const useGetChildrenToArray = ({
+  children,
+  filterLevel,
+}: {|
+  children: Node,
+  filterLevel: 'main' | 'nested',
+|}): $ReadOnlyArray<Node> => {
+  let hasFirstNavigationItem = false;
```

Comment:
I think this var name is not the clearest, is it more like "foundFirstNavigationItem"?

Reply:
DONE